internal/resolver/dns/dns_resolver.go

```diff
@@ -354,13 +354,22 @@ func (d *dnsResolver) lookup() (*resolver.State, error) {
 // If addr is an IPv6 address, return the addr enclosed in square brackets and
 // ok = true.
 func formatIP(addr string) (addrIP string, ok bool) {
-	ip := net.ParseIP(addr)
+	// Split the address into IP and zone parts if it contains a zone identifier
+	ipStr := addr
```

Comment:
hmmm...we should not be doing any parsing from our side. Is net.ParseIP() not parsing the address correctly?